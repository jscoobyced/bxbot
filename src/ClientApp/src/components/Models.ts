export class Constants {
    public static readonly StaticContent = (process.env.mode === 'development') ? '' : '/dist';

    public static readonly ImagePath = `${Constants.StaticContent}/images/`;
}

export interface SelectOption {
    value: number;
    text: string;
}
