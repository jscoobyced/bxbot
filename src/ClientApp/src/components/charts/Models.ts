export interface Pairing {
    timestamp: number;
    low: number;
    high: number;
    close: number;
    open: number;
}

export interface CandleChartPageState {
    pairings: Pairing[];
    loading: boolean;
}

export interface CandleChartProps {
    currency: string;
    element: string;
}