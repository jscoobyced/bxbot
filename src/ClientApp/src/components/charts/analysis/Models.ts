import { Pairing, Pattern } from "../Models";

export interface IAnalyser {
    analyse(pairings: Pairing[], index: number): Pattern;
}