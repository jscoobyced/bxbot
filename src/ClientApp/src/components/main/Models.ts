import { NavigationMenuContent } from "../../services/Models";

export interface HeaderProps {
    navigationMenuContent: NavigationMenuContent;
}
