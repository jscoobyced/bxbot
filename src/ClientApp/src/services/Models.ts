export enum Language {
    English,
    French
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
}

export interface NavigationMenuContent {
    home: string;
    websiteName: string;
    cryptoCurrency: string;
    candleChart: string;
    about: string;
}

export interface HomeContent {
    articleTitle: string;
    article: string;
}