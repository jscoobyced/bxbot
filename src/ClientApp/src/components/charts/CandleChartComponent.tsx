import * as React from 'react';
import { CandleChart } from './CandleChart';
import { CandleChartDrawer } from './CandleChartDrawer';
import { DataFormatter } from './DataFormatter';
import { GoogleChartInitializer } from './GoogleChartInitializer';
import { CandleChartComponentState } from './Models';

export class CandleChartComponent extends React.Component<CandleChartComponentState> {
    private static readonly ChartElement = "candle_div";
    private candleChart: CandleChartDrawer = new CandleChartDrawer([], "");
    private readonly bollingerSize = 20;
    private readonly dataFormatter: DataFormatter = new DataFormatter();
    private chartData: any;

    constructor(props: CandleChartComponentState) {
        super(props);
    }

    public componentDidUpdate() {
        const { pairings } = this.props;
        if (pairings !== []) {
            this.chartData = this.dataFormatter.formatData(pairings, this.bollingerSize);
            if (!GoogleChartInitializer.IsReady()) {
                new GoogleChartInitializer().Init(this.drawChart);
            } else {
                this.drawChart();
            }
        }
    }

    public render(): JSX.Element {
        return <CandleChart
            element={CandleChartComponent.ChartElement}
            currency={this.props.currency} />;
    }

    private readonly drawChart = () => {
        this.candleChart = new CandleChartDrawer(this.chartData, CandleChartComponent.ChartElement);
        this.candleChart.drawChart();
    }
}
