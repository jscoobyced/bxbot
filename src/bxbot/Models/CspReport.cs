using Newtonsoft.Json;

namespace bxbot
{
    public class CspReport
    {
        [JsonProperty(PropertyName = "document-uri")]
        public string DocumentUri { get; set; }

        [JsonProperty(PropertyName = "referrer")]
        public string Referrer { get; set; }

        [JsonProperty(PropertyName = "violated-directive")]
        public string ViolatedDirective { get; set; }

        [JsonProperty(PropertyName = "effective-directive")]
        public string EffectiveDirective { get; set; }

        [JsonProperty(PropertyName = "original-policy")]
        public string OriginalPolicy { get; set; }

        [JsonProperty(PropertyName = "blocked-uri")]
        public string BlockedUri { get; set; }

        [JsonProperty(PropertyName = "status-code")]
        public int StatusCode { get; set; }
    }
}