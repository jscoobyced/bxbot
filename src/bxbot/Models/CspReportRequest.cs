using Newtonsoft.Json;

namespace bxbot
{
    public class CspReportRequest
    {
        [JsonProperty(PropertyName = "csp-report")]
        public CspReport CspReport { get; set; }
    }
}