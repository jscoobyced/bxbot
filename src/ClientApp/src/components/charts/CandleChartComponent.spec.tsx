import { shallow } from 'enzyme';
import * as React from 'react';
import { CandleChartComponent } from './CandleChartComponent';
import { Pairing } from './Models';

test('CandleChartComponent isn\'t changed when empty', () => {
    const candleChartComponent = shallow(<CandleChartComponent pairings={[]} currency={'BTC'} loading={false} />);
    expect(candleChartComponent).toMatchSnapshot();
});

test('CandleChartComponent isn\'t changed when not empty', () => {
    const pairings: Pairing[] = [{ close: 0, high: 0, low: 0, open: 0, timestamp: 0 }];
    const candleChartComponent = shallow(<CandleChartComponent pairings={[]} currency={'BTC'} loading={false} />);
    candleChartComponent.setProps({ pairings: pairings });
    expect(candleChartComponent).toMatchSnapshot();
});