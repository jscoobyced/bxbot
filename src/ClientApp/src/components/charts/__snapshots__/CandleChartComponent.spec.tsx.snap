// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandleChartComponent isn't changed when empty 1`] = `
<div
  id="candle_div"
/>
`;

exports[`CandleChartComponent isn't changed when not empty 1`] = `
<div
  id="candle_div"
/>
`;
